src/Take.Blip.Client/Extensions/Tunnel/EnvelopeExtensions.cs

```diff
@@ -7,8 +7,16 @@ public static class EnvelopeExtensions
     {
         public static bool TryGetTunnelFromEnvelope<T>(this T envelope, out Takenet.Iris.Messaging.Resources.Tunnel tunnelInformation) where T : Envelope
         {
-            if (envelope.From?.Domain != null &&
-                envelope.From.Domain.Equals(TunnelExtension.TunnelAddress.Domain, StringComparison.OrdinalIgnoreCase) &&
+            Node fromNode = envelope.From;
+
+            if (envelope.Metadata != null && envelope.Metadata.ContainsKey(Constants.ORIGINAL_SUBFLOW_REDIRECT_FROM))
+            {
+                envelope.Metadata.TryGetValue(Constants.ORIGINAL_SUBFLOW_REDIRECT_FROM, out string originalFrom);
```

Comment:
You can use the `.TryGetValue` inside the if instead of checking with the `.ContainsKey`.

Reply:
done